### PR TITLE
Median of arrays in c

### LINKED ISSUE #4961: Median of two sorted arrays in C
## 💥 Proposal

Given two sorted arrays nums1 and nums2 of size m and n respectively, return the median of the two sorted arrays.
I am a GSSOC participant and wan t to contribute in C.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by Amit366**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Median of two sorted arrays in C

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

fixes: #4961

### CODE REVIEW

**Review by atarax665 — CHANGES_REQUESTED** [review]
Changes requested:
- Rename the file name  from ```Median of two sorted arrays.c``` to  ```Median_of_two_sorted_arrays.c``` and make appropriate changes to thge README.md aswell.
- Remove unnecessary blank lines
- Add proper input-output statements to make the program more interactive.
- Remove variables ```sum and s``` as they aren't being used in the program

**Comment by Amit366** [comment]
@atarax665 updated. Please check.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Use cpplint it will guide you how to format code properly for reference you can use this one [here](https://cpplint.appspot.com/).

**Comment by Amit366** [comment]
@ankitaggarwal23 updated. Please review.

**Comment by Amit366** [comment]
@ankitaggarwal23 please review it.

**Comment by Amit366** [comment]
@HarshCasper please review it.

File: C/cp/Median_of_two_sorted_arrays.c [hunk 1]
```diff
@@ -0,0 +1,103 @@
+// Given two sorted arrays nums1 and nums2 of size m and n
+// respectively, return the median of the two sorted arrays
```
Comment (HarshCasper) [inline C/cp/Median_of_two_sorted_arrays.c]: Use multi-line comments

File: C/cp/Median_of_two_sorted_arrays.c [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+#include<stdio.h>
+#include<stdlib.h>
```
Comment (HarshCasper) [inline C/cp/Median_of_two_sorted_arrays.c:4]: Add a new line after this

File: C/cp/Median_of_two_sorted_arrays.c [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+int main() {
+  int n, m;
+  // accepting the sizes
+  printf("Enter sizes of both arrays: ");
+  scanf("%d %d", & n, & m);
+  int * arr1 = (int *) malloc(sizeof(int) * n);
+  int * arr2 = (int *) malloc(sizeof(int) * m);
+  // accepting the array elements
+  printf("Enter elements of first array: ");
+  for (int i = 0; i < n; i++) {
```
Comment (HarshCasper) [inline C/cp/Median_of_two_sorted_arrays.c:17]: Add a new line after this

File: C/cp/Median_of_two_sorted_arrays.c [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    scanf("%d ", arr1 + i);
+  }
+  printf("Enter elements of second array: ");
```
Comment (HarshCasper) [inline C/cp/Median_of_two_sorted_arrays.c:20]: Add a new line after this

File: C/cp/Median_of_two_sorted_arrays.c [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  for (int i = 0; i < m; i++) {
+    scanf("%d ", arr2 + i);
+  }
+  // print the median
+  printf("Median of two arrays is %f",
+         findMedianSortedArrays(arr1, n, arr2, m));
+  return 0;
+}
```
Comment (HarshCasper) [inline C/cp/Median_of_two_sorted_arrays.c:29]: Add a new line after this

File: C/cp/Median_of_two_sorted_arrays.c [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        k++;
+      } else if (arr2[k] < arr1[j]) {
+        arr3[i] = arr2[k];
+        k++;
+      } else if (arr1[j] <= arr2[k]) {
+        arr3[i] = arr1[j];
+        j++;
+      }
+    }
+  }
```
Comment (HarshCasper) [inline C/cp/Median_of_two_sorted_arrays.c:66]: Add a new line after this

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by Amit366** [comment]
@HarshCasper initially I added the new lines but @ankitaggarwal23 said to format it and in the cpplint the extra lines where showing errors.

**Comment by Amit366** [comment]
@HarshCasper please review

**Comment by Amit366** [comment]
@HarshCasper please review it.

**Review by HarshCasper — APPROVED** [review]

**Comment by Amit366** [comment]
@HarshCasper could you please add the labels??

**Pull request merged**